extensions/functions_string.yaml

```diff
@@ -251,6 +251,104 @@ scalar_functions:
             name: "replacement"
             description: The replacement string.
         return: "varchar<L1>"
+  -
+    name: regexp_replace
+    description: >-
+      Search a string for a substring that matches a given regular expression pattern and replace
+      it with a replacement string. The regular expression pattern should follow the
+      International Components for Unicode implementation (https://unicode-org.github
+      .io/icu/userguide/strings/regexp.html). The occurrence of the pattern to be replaced is
```

Comment:
Awesome, I didn't know that one. Nice and concise, and an actual standard rather than the usual "it behaves like the matcher in this random tool or language".